lib/src/presentation/bloc/bloc_movie_list.dart

```diff
@@ -0,0 +1,72 @@
+import 'dart:async';
+
+import 'package:either_dart/either.dart';
+
+import '../../core/util/movie_list_enum.dart';
+import '../../domain/entity/movie_entity.dart';
+import '../../domain/usecase/implementation/now_playing_movies_usecase.dart';
+import '../../domain/usecase/implementation/popular_movies_usecase.dart';
+import '../../domain/usecase/implementation/top_rated_movies_usecase.dart';
+import '../../domain/usecase/implementation/up_coming_movies_usecase.dart';
+import 'bloc.dart';
+
+class BlocMovieList implements Bloc {
+  BlocMovieList({
+    required this.nowPlayingMoviesUseCase,
+    required this.upComingMoviesUseCase,
+    required this.popularMoviesUseCase,
+    required this.topRatedMoviesUseCase,
+  });
+
+  final PopularMoviesUseCase popularMoviesUseCase;
+  final TopRatedMoviesUseCase topRatedMoviesUseCase;
+  final UpComingMoviesUseCase upComingMoviesUseCase;
+  final NowPlayingMoviesUseCase nowPlayingMoviesUseCase;
+
+  final StreamController<List<Movie>> _movieList =
+      StreamController<List<Movie>>();
+
+  Stream<List<Movie>> get allMovies => _movieList.stream;
+
+  void fetchMovies(MovieEndpoint movieEndpoint) async {
+    switch (movieEndpoint) {
+      case MovieEndpoint.POPULAR:
+        await popularMoviesUseCase.execute().fold(
+              (error) => _movieList.sink.addError(error),
+              (movieList) => _movieList.sink.add(movieList),
+            );
+        break;
+      case MovieEndpoint.TOP_RATED:
+        await topRatedMoviesUseCase.execute().fold(
+              (error) => _movieList.sink.addError(error),
+              (movieList) => _movieList.sink.add(movieList),
+            );
+        break;
+      case MovieEndpoint.NOW_PLAYING:
+        await nowPlayingMoviesUseCase.execute().fold(
+              (error) => _movieList.sink.addError(error),
+              (movieList) => _movieList.sink.add(movieList),
+            );
+        break;
+      case MovieEndpoint.UP_COMING:
+        await upComingMoviesUseCase.execute().fold(
+              (error) => _movieList.sink.addError(error),
+              (movieList) => _movieList.sink.add(movieList),
+            );
+        break;
+      default:
+        await popularMoviesUseCase.execute().fold(
+              (error) => _movieList.sink.addError(error),
+              (movieList) => _movieList.sink.add(movieList),
+            );
```

Comment:
duplicated code again, let's finde a wat to abstract this code

Reply:
fixed